netutils/password.py

```diff
@@ -197,23 +142,36 @@ def encrypt_type7(unencrypted_password, salt=None):
         salt (str, optional): A random number between 0 and 15 that can be set by the operator. Defaults to random generated one.
 
     Returns:
-        string: The encrypted password.
+        str (optional): The encrypted password.
 
     Example:
         >>> from netutils.password import encrypt_type7
-        >>> encrypt_type5("cisco")  # doctest: +SKIP
-        '$1$ZLGo$J.gAGxS2wqO96drs0Cith/'
+        >>> encrypt_type7("cisco", 11)
+        '110A1016141D'
         >>>
     """
+    # max length of password for encrypt t7 is 25
+    if len(unencrypted_password) > 25:  # nosec
+        raise ValueError("Password must not exceed 25 characters.")
+
+    key_hex = []
+    # the same key string is used in decrypt_type7 for the reverse operation
+    for char in "dsfd;kfoA,.iyewrkldJKDHSUBsgvca69834ncxv9873254k;fg87":
```

Comment:
This is the same as the XLAT variable, I am not sure if it needs to be changed 

```
>>> for char in "dsfd;kfoA,.iyewrkldJKDHSUBsgvca69834ncxv9873254k;fg87":
...   print(hex(ord(char)))
...
0x64
0x73
0x66
0x64
0x3b
0x6b
0x66
0x6f
0x41
0x2c
0x2e
0x69
0x79
0x65
0x77
0x72
0x6b
0x6c
0x64
0x4a
0x4b
0x44
0x48
0x53
0x55
0x42
0x73
0x67
0x76
0x63
0x61
0x36
0x39
0x38
0x33
0x34
0x6e
0x63
0x78
0x76
0x39
0x38
0x37
0x33
0x32
0x35
0x34
0x6b
0x3b
0x66
0x67
0x38
0x37
```

Reply:
Added with string quotes, otherwise it does not work properly in my implementation.